A3A/addons/ultimate/patches/cfgVehicles.hpp

```diff
@@ -261,6 +261,928 @@ class CfgVehicles {
             previewHeight = 30;
         };
     };
+
+    // Custom classes for Antistasi Units / Vehicles in Eden / Zeus
+    class SoldierWB;
+    class I_Soldier_base_F : SoldierWB {
+        class EventHandlers;
+    };
+    class A3U_Rebels_Base: I_Soldier_base_F {
+        scope = 1;
+        scopeCurator = 1;
+        scopeArsenal = 1;
+        faction = "A3U_Rebels";
+        //icon = "";
+        //picture = "";
+        //role = "";
+        //nameSound = "";
+        //textSingular = "";
+        //textPlural = "";
+        uniformClass = "";
+        backpack = "";
+        items[] = {};
+        respawnItems[] = {};
+        linkedItems[] = {};
+        respawnLinkedItems[] = {};
+        weapons[] = {"Throw", "Put"};
+        respawnWeapons[] = {"Throw", "Put"};
+        magazines[] = {};
+        respawnMagazines[] = {};
+        //genericNames = "";
+        //identityTypes[] = {};
+        class EventHandlers {
+            class ADDON {
+                PostInit = "params ['_unit']; [_unit, false] call A3A_fnc_fiaInit;";
+            };
+        };
```

Comment:
The base class can do all the repetitive work you're doing in the child classes:

```suggestion
        class EventHandlers {
            class ADDON {
                PostInit = QUOTE([ARR_2(_this#0,false)] call A3A_fnc_fiaInit);
                init = QUOTE((_this#0) setVariable[ARR_2(QUOTE(QUOTE(unitType)),getText(configOf(_this#0) >> QQGVAR(unitType)))]);
            };
        };
```

Reply:
Whaaaaaaaa.

I'ma need some schooling on this GVAR stuff. I did not like the init / postinit EHs either, but couldn't figure out how to set an arbitrary var on the units in config.

Does GVAR do the same thing in config as `setVariable` in sqf? Or similar idea anyway? That's definitely a lot cleaner if so. I mainly just didn't want to write a whole separate init func for the units and reuse the existing fiaInit / NATOinit which rely on variables.

Reply:
> Does GVAR do the same thing in config as setVariable in sqf?

No, not at all. The macro is used to prefix (global) variables or things that could conflict with other mods. `GVAR(foo)` is basically `A3A_foo` in this case.

In [this suggestion](https://github.com/Antistasi-Ultimate-Community/A3-Antistasi-Ultimate/pull/681/files#r2513712053), `GVAR(unitType)` (i.e. `A3A_unitType`) is a property for the child class. Since the child class uses its parent class' `init` code (from this suggestion), it can read the property.

Reply:
Ok, I think I get it.

Will commit the changes for the rebels, and then play with doing something similar for the occupants / enemy factions. Got an idea of how to clean those up in a similar vein.